python_scripts/ensemble_gradient_boosting.py

```diff
@@ -199,10 +199,11 @@ def generate_data(n_samples=50):
 # and compare it with the true value.
 
 # %%
-print(f"True value to predict for f(x={data_max:.3f}) = {target_true:.3f}")
+print(f"True value to predict for"
+      f"f(x={data_max['Feature'].iloc[0]:.3f}) = {target_true:.3f}")
```

Comment:
Maybe you can define `x_sample = data_max['Feature'].iloc[0]` (for example the line right after `sample` is defined?) and reuse it in the f-strings

Reply:
Good idea, it's done!